### DISCUSSION TITLE
How to be user www-data?

### DISCUSSION THREAD

**Discussion by Emporea** [coder/code-server#5692] [Q&A]
I have a mounted storage whose owner is www-data(33) and this shouldnt be changed.
Thus I want to use code-server with the uid 33.

Should I change the uid of coder (1000) to 33 or can I do something else?

I tried alot of stuff but i keep getting errors.

I tried setting `USER 33` in my Dockerfile. If I do this I keep getting errors when install code-server extensions about permissions i dont have about /var/www or something. Also the user is still called coder ( which would be fine if the uid was 33).

Then I tried setting `user: 33:33` in my docker-compose.yml file. 
This does work to some extent, atleast the fixuid script gets started, but the container crashes after about 1 minute without serving anything on the web nor producing any logs (atleast not the docker container).

I tried setting DOCKER_USER=www-data, but this doesnt change anything. The default home directory is still /home/coder. 

I am confused.

How can I be a user with uid 33, while being able to already install some extensions in the Dockerfile?

Thanks for help.

**Comment by jsjoeio** [comment]
Seems like a good question for @code-asher

**Comment by code-asher** [comment]
I believe we keep the home directory as `/home/coder` because we are not able to move bind mounts.  But if you use `-u "33:33"`  the permissions should be correct.  You might just need to mount your directory over `/home/coder` like `-v /path/to/www-data:/home/coder`.

**Reply by code-asher** [reply]
But it looks like you have already tried that so I am not sure.  I can give it a try some time this week.

**Reply by Emporea** [reply]
This i would appreciate!

**Reply by code-asher** [reply]
Initial setup:

```console
$ whoami && id
coder
uid=1000(coder) gid=1000(coder) groups=1000(coder),999(docker)
$ sudo mkdir /var/www
$ sudo chown -R www-data: /var/www
```

This seems to work for me to run as `www-data` with my data directory set to `/var/www`:

``` console
docker run -it --name code-server -p 127.0.0.1:8080:8080 -e SHELL=/bin/bash -v "/var/www:/var/www" --user www-data codercom/code-server:latest --user-data-dir /var/www
```

I set `SHELL` since `www-data` by default has no shell.
